binary/relations.proto

```diff
@@ -176,5 +188,6 @@ message Rel {
       JoinRel join = 6;
       ProjectRel project = 7;
       SetRel set = 8;
+      NamedRel named = 9;
```

Comment:
I guess one thing that is potentially awkward is the ability to compose `NamedRel`s, which seems valid, but not very useful.

Reply:
That's why I don't think it belongs in the Rel union. I'm inclined to elevate, similar to NamedStruct. It also removes the inclusion of common, which doesn't really belong in this "marker".

Reply:
One other idea. Maybe we call this RelRoot. That's what Calcite [calls it](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/RelRoot.java). It has some other cruft but it is basically a rel + field names (called "validated row type" in calcite).

Reply:
`RelRoot` does seem like an apt name. Let's go with that. I'll move it out as well.